### PR TITLE
Remove kcp adoption test

### LINKED ISSUE #2596: Remove KCP adoption test
The below test was added to ensure existing clusters with control plane machines before KCP (KubeadmControlPlane) was introduced could be adopted by a KCP object: 

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/test/e2e/capi_test.go#L158

All new clusters should be using KCP for clusters built since ~ v1alpha4 so we should deprecate (remove) the test from the CAPI e2e test suite, and remove the associated e2e test templates (https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/main/test/e2e/data/infrastructure-azure/v1beta1/cluster-template-kcp-adoption) and all references to it.

### PR DESCRIPTION
**Pull request by willie-yao**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Removes KCP adoption tests and test templates since all new clusters should be using KCP.

**Which issue(s) this PR fixes**:
Fixes #2596 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/data/infrastructure-azure/v1beta1/bases/cluster-with-kcp.yaml [hunk 1]
```diff
@@ -1,143 +0,0 @@
----
```
Comment (mboersma) [inline test/e2e/data/infrastructure-azure/v1beta1/bases/cluster-with-kcp.yaml:1]: Isn't this file still referred to by other test templates, like `cluster-template-kcp-remediation`?

**Review by mboersma — COMMENTED** [review]
/lgtm

File: test/e2e/data/infrastructure-azure/v1beta1/bases/cluster-with-kcp.yaml [hunk 2]
```diff
@@ -140,4 +140,4 @@ spec:
       - - LABEL=etcd_disk
         - /var/lib/etcddisk
   replicas: ${CONTROL_PLANE_MACHINE_COUNT}
-  version: ${KUBERNETES_VERSION}
+  version: ${KUBERNETES_VERSION}
```
Comment (CecileRobertMichon) [inline test/e2e/data/infrastructure-azure/v1beta1/bases/cluster-with-kcp.yaml:143]: nit: re-add the newline, this file shouldn't change

**Review by mboersma — COMMENTED** [review]
/lgtm

**Pull request merged**